fix for issue hpilo_ca sign needs a DNS alias. #80

duncanwebb opened the pull request:
I've had some help putting the changes into git so that you can pull them out :smile:

This version also fixes that openssl.cnf file generation so that variables in ca section the ilo.conf will be used to generate the file during the init command. I've now updated the certificates on about 80 blades of type G1, G6, G7, G8 and G9.

The sign command looks like this:
./hpilo_ca --login=Administrator --password=$password sign 10.30.25.16 10.30.25.17 10.30.25.18
and will return a list of hosts that are generating the CSR.

Thanks for this toolkit it is great.

seveas commented:
Thank you! I probably won't merge this as is but turn it into a set of smaller patches, so please be patient while I do so.

seveas commented:
I've finally found some time to look at this again. In commit 5f6a076 I am taking an alternative approach, but with the same outcome: SAN's are added to certificates and there's retry logic for when the certificate takes a while to generate.

I also like the idea of allowing country etc. in the config file, so I added that too.

Thanks!

duncanwebb commented:
Thanks for spending the time doing this. There are a couple of things that I changed since then and that was the 
default_md       = sha1
to 
default_md       = sha265

Also added "localityName           = optional" to the req_policy.

The first change is to prevent errors and the second is to include the locality in the certificate so it is shown in the iLO.

seveas commented:
Thanks, both these changes make sense, so I've added those as well and they'll be in the next version.
